contracts/test/mocks/BondingVotesERC5805Harness.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+import "../../bonding/BondingVotes.sol";
+import "./GenericMock.sol";
+
+/**
+ * @dev This is a tets utility for unit tests on the ERC5805 functions of the BondingVotes contract. It overrides the
+ * functions that should be used to derive the values returned by the ERC5805 functions and checks against those.
+ */
+contract BondingVotesERC5805Harness is BondingVotes {
```

Comment:
Not sure I understand how this is being used from a quick glance. If you link to the relevant areas that this contract is used and the test cases written for it can prob comment on that.

Reply:
Ah I see. 

Hmm it seems that the bulk of the core logic in BondingVotes right now *is* the checkpointing logic and then all the ERC-5805 functions are either fairly simple or rely on calling functions that implement the checkpointing logic. While the module approach might make testing a bit easier, it does feel like it would end up breaking up into a module (i.e. BondingVotesERC5805) that doesn't do much that just relies on abstract functions that need to be implemented in the child contract. I feel ok about the current testing approach..., but I would just document the rationale in the tests themselves that use the harness contract.